pulsar-websocket/src/main/java/org/apache/pulsar/websocket/AbstractWebSocketHandler.java

```diff
@@ -55,44 +56,56 @@ public AbstractWebSocketHandler(WebSocketService service, HttpServletRequest req
         request.getParameterMap().forEach((key, values) -> {
             queryParams.put(key, values[0]);
         });
-    }
 
-    @Override
-    public void onWebSocketConnect(Session session) {
-        super.onWebSocketConnect(session);
-        log.info("[{}] New WebSocket session on topic {}", session.getRemoteAddress(), topic);
+        checkAuth(response);
```

Comment:
Right now, `onWebSocketConnect()` if request fail due to `auth&auth` then we [close](https://github.com/apache/incubator-pulsar/blob/master/pulsar-websocket/src/main/java/org/apache/pulsar/websocket/AbstractWebSocketHandler.java#L75) the connection.
It seems we are not doing it now.  It seems if `checkAuth()` fails then we still continue to create producer and consumer and client will be able to publish and consume?

Reply:
It should get a `401` error before reaching the `onConnect()` state.

`onConnect()` happens when the client receives the `101 Upgrade` HTTP response from proxy. At that point it will start talking websocket with the proxy. 

By doing the validation before, we just return 401 or 403 instead of 101

Reply:
I see. you mean `response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Failed to authenticate");` will not allow socket to upgrade the request and it won't be able to connect.?
and if we get exception at line#77 then does it mean proxy failed to send error-response to client.? In that case, can client connect ? or we can close the connection at that line?
```
             try {
                    response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Failed to authenticate");
                } catch (IOException e1) {
                    log.warn("Failed to send error: {}", e1.getMessage());
                }
```

Reply:
Yes, at that point we don't have the websocket session object yet, so we cannot do the regular `close()` method. The only thing we have is the `response` context and if we cannot write to it I think the only reason must be that the client disconnected.